### PR TITLE
F51-462, F51-275, F51-102 - add to cart validation & row flexing

### PR DESCRIPTION
**Pull request by robertsoniv**
See commits for details.

### CODE REVIEW

**Comment by robertsoniv** [comment]
I will be pushing one more task to this PR before re-opening

File: src/app/common/directives/oc-product-card/oc-product-card.js [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
-            currentOrder: '<',
-            lineitemlist: '<'
-        }
+    .controller('ProductCardCtrl', OrderCloudProductCardController)
+    .directive('ocProductCard', function() {
+        return {
+            templateUrl: 'common/directives/oc-product-card/oc-product-card.html',
+            controller: 'ProductCardCtrl',
+            controllerAs: 'productCard',
+            replace: true,
```
Comment (crhistianr) [inline src/app/common/directives/oc-product-card/oc-product-card.js]: This has been deprecated for a while now.

File: src/app/common/directives/oc-quantity-input/oc-quantity-input.js [hunk 2]
```diff
@@ -8,13 +8,20 @@ function OCQuantityInput($log, $state, toastr, OrderCloudSDK) {
...
             label: '@',
             order: '=',
-            onUpdate: '&'
+            onUpdate: '&',
+            size: '@'
         },
+        require: '^?ocPrettySubmit',
         templateUrl: 'common/directives/oc-quantity-input/oc-quantity-input.html',
         replace: true,
-        link: function (scope) {
```
Comment (crhistianr) [inline src/app/common/directives/oc-quantity-input/oc-quantity-input.js]: same as above

**Comment by crhistianr** [comment]
javascript looks ok except for the `replace:true` thing

**Comment by robertsoniv** [comment]
replace: true is what makes the styles work.  without it I will have to change html.

File: src/app/common/directives/oc-quantity-input/oc-quantity-input.js (hunk 2 above)
Comment (olsonkr) [inline src/app/common/directives/oc-quantity-input/oc-quantity-input.js]: It looks like it's only deprecated on 2.0+. We're still on 1.6.0.

**Pull request merged**